### PR TITLE
Add documentation for SEM

### PR DESCRIPTION
**Pull request by morgo**
### What is changed, added or deleted? (Required)

This adds the docs for https://github.com/pingcap/tidb/blob/master/docs/design/2021-03-09-security-enhanced-mode.md

Which was merged in a series of PRs to master. The exact restrictions may evolve over time, so the wording is kept to use simple language to say "restrictions include". It is intended for use in cloud environments where the "root" privilege is considered too powerful.

### Which TiDB version(s) do your changes apply to? (Required)

This only affects master.

- [x] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/24416
https://github.com/pingcap/tidb/pull/24412
https://github.com/pingcap/tidb/pull/24279
https://github.com/pingcap/tidb/pull/23978

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/blob/master/docs/design/2021-03-09-security-enhanced-mode.md

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by bb7133** [comment]
PTAL @SunRunAway

File: system-variables.md [hunk 1]
```diff
@@ -427,6 +427,17 @@ Constraint checking is always performed in place for pessimistic transactions (d
 - Default value: ON
 - This variable controls whether to record the execution information of each operator in the slow query log.
 
+### tidb_enable_enhanced_security
```
Comment (SunRunAway) [inline system-variables.md:430]: Is this variable readonly?

Comment (SunRunAway) [reply system-variables.md:430]: @morgo PTAL

Comment (morgo) [reply system-variables.md:430]: Yes it is. Or more specifically it is `Scope: None`, which is the same as read-only.

Comment (SunRunAway) [reply system-variables.md:430]: > TiDB system variables behave similar to MySQL with some differences, in that settings might apply on a SESSION, INSTANCE, or GLOBAL scope, or on a scope that combines SESSION, INSTANCE, or GLOBAL.

> * Changes to GLOBAL scoped variables only apply to new connection sessions with TiDB. Currently active connection sessions are not affected. These changes are persisted and valid after restarts.
> * Changes to INSTANCE scoped variables apply to all active or new connection sessions with the current TiDB instance immediately after the changes are made. Other TiDB instances are not affected. These changes are not persisted and become invalid after TiDB restarts.

The document does not mention any about what `Scope: None` means. Cloud you add some explanations?

Comment (morgo) [reply system-variables.md:430]: I agree with you. I will change it tomorrow to say Read Only instead.

(There are some minor differences, but they don't matter to users.)

Comment (morgo) [reply system-variables.md:431]: This is improved in https://github.com/pingcap/docs/pull/5647/commits/c8758d7b8c130ed8c52aa331e27110c383fed166 , PTAL again thx :-)

**Review by SunRunAway — APPROVED** [review]

**Comment by TomShawn** [comment]
Follow-up: Minor wording issue

File: system-variables.md [hunk 2]
```diff
@@ -427,6 +428,17 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+
+- Scope: NONE
+- Default value: OFF
+- This variable indicates if the TiDB server you are connected to has Security Enhanced Mode (SEM) enabled, and can not be changed without restarting the TiDB server.
```
Comment (TomShawn) [inline system-variables.md:435]: @morgo Can we say "To change its value, you need to restart the TiDB server for the change to be effective"?

**Pull request merged**